stag-library-compiler/src/main/java/com/vimeo/stag/processor/utils/ElementUtils.java

```diff
@@ -72,6 +76,42 @@ public static String getPackage(@NotNull TypeMirror type) {
     }
 
     /**
+     * Get the annotation applied of the specified type if it has been applied to the provided
+     * element.  This method will check the supplied element and work up the type hierarchy,
+     * returning the first annotation instance found.
+     *
+     * @param annotationType annotation type to search for
+     * @param element element to analyze
+     * @param <T> annotation type
+     * @return annotation instance or {@code null} if not found
+     */
+    @Nullable
+    public static <T extends Annotation> T findAnnotation(
+            @NotNull Class<T> annotationType, @NotNull Element element) {
+        T annotatedElement = element.getAnnotation(annotationType);
+        if (annotatedElement != null)
+            return annotatedElement;
+
+        if (element instanceof TypeElement) {
+            TypeElement typeElement = (TypeElement) element;
+            TypeMirror superTypeMirror = typeElement.getSuperclass();
+            if (superTypeMirror != null) {
+                Element superTypeElement = TypeUtils.getElementFromTypeMirror(superTypeMirror);
+                return superTypeElement == null ? null : findAnnotation(annotationType, superTypeElement);
+            }
+        }
```

Comment:
Yeah I think it's fine to leave for right now. The fact that there's an open PR for the removal of gson adapter key is kind of muddying some of this logic, but we can remove it if it's necessary in that PR. Fine as is.